onnxruntime/core/session/inference_session.cc

```diff
@@ -955,12 +1020,45 @@ class InferenceSession::Impl {
     return Status::OK();
   }
 
+  Status ValidateLevel(unsigned int level) const {
+    auto t_level = (unsigned int)std::pow(2, level);
+    if (t_level < TransformerLevel::MaxTransformerLevel) {
+      return Status::OK();
+    }
+    return Status(ONNXRUNTIME, FAIL, "Unsupported level " + level);
+  }
+
+  void SetTransformerContext(uint32_t& transformers_enabled, std::vector<TransformerLevel>& all_levels, const uint32_t& level) {
```

Comment:
>transformers_enabled [](start = 39, length = 20)

transformers_enabled -> levels_enabled would be clearer, esp. given you have 'all_levels'